neuro_san/client/direct_agent_session_factory.py

```diff
@@ -50,29 +51,53 @@ def create_session(self, agent_name: str, use_direct: bool = False,
                        metadata: Dict[str, str] = None) -> AgentSession:
         """
         :param agent_name: The name of the agent to use for the session.
+                This name can be something in the manifest file (with no file suffix)
+                or a specific full-reference to an agent network's hocon file.
         :param use_direct: When True, will use a Direct session for
                     external agents that would reside on the same server.
         :param metadata: A grpc metadata of key/value pairs to be inserted into
                          the header. Default is None. Preferred format is a
                          dictionary of string keys to string values.
         """
 
-        factory = ExternalAgentSessionFactory(use_direct=use_direct)
-        tool_factory: ServiceToolFactoryProvider =\
-            ServiceToolFactoryProvider.get_instance()
-        tool_registry_provider: AgentToolFactoryProvider =\
-            tool_factory.get_agent_tool_factory_provider(agent_name)
-        tool_registry: AgentToolRegistry = tool_registry_provider.get_agent_tool_factory()
+        tool_registry: AgentToolRegistry = self.get_agent_tool_registry(agent_name)
 
         llm_factory: ContextTypeLlmFactory = MasterLlmFactory.create_llm_factory()
         base_tool_factory: ContextTypeBaseToolFactory = MasterBaseToolFactory.create_base_tool_factory()
         # Load once now that we know what tool registry to use.
         llm_factory.load()
         base_tool_factory.load()
 
+        factory = ExternalAgentSessionFactory(use_direct=use_direct)
         invocation_context = SessionInvocationContext(factory, llm_factory, base_tool_factory, metadata)
         invocation_context.start()
         session: DirectAgentSession = DirectAgentSession(tool_registry=tool_registry,
                                                          invocation_context=invocation_context,
                                                          metadata=metadata)
         return session
+
+    def get_agent_tool_registry(self, agent_name: str) -> AgentToolRegistry:
+        """
+        :param agent_name: The name of the agent whose AgentToolRegistry we want to get.
+                This name can be something in the manifest file (with no file suffix)
+                or a specific full-reference to an agent network's hocon file.
+        :return: The AgentToolRegistry corresponding to that agent.
+        """
+
+        if agent_name is None or len(agent_name) == 0:
+            return None
+
+        tool_registry: AgentToolRegistry = None
+        if agent_name.endswith(".hocon") or agent_name.endswith(".json"):
+            # We got a specific file name
+            restorer = AgentToolRegistryRestorer()
+            tool_registry = restorer.restore(file_reference=agent_name)
```

Comment:
If the agent referenced is a hocon file, load it to use as our agent.

Reply:
This will only work if agent name is from "original" registry, right?
Otherwise service routing will fail.

Reply:
No. it can be from anywhere.